packages/runtime/container-runtime/src/test/containerRuntime.spec.ts

```diff
@@ -2564,35 +2565,55 @@ describe("Runtime", () => {
 					// eslint-disable-next-line @typescript-eslint/dot-notation
 					containerRuntime["channelCollection"]["contexts"].get("missingDataStore");
 				assert(missingDataStoreContext !== undefined, "context should be there");
-				// Add ops to this context.
-				const messages = [
-					{ sequenceNumber: 1 },
-					{ sequenceNumber: 2 },
-					{ sequenceNumber: 3 },
-					{ sequenceNumber: 4 },
-				];
-				// eslint-disable-next-line @typescript-eslint/dot-notation
-				missingDataStoreContext["pending"] = messages as ISequencedDocumentMessage[];
+				missingDataStoreContext.processMessages(
+					[
+						{
+							message: { sequenceNumber: 1 } as unknown as ISequencedDocumentMessage,
+							localOpMetadata: undefined,
+						},
+						{
+							message: { sequenceNumber: 2 } as unknown as ISequencedDocumentMessage,
+							localOpMetadata: undefined,
+						},
+						{
+							message: { sequenceNumber: 3 } as unknown as ISequencedDocumentMessage,
+							localOpMetadata: undefined,
+						},
+						{
+							message: { sequenceNumber: 4 } as unknown as ISequencedDocumentMessage,
+							localOpMetadata: undefined,
+						},
+					],
+					false,
+				);
 
 				// Set it to seq number of partial fetched snapshot so that it is returned successfully by container runtime.
 				(containerContext.deltaManager as any).lastSequenceNumber = 2;
 
 				let opsProcessed = 0;
 				let opsStart: number | undefined;
-				missingDataStoreRuntime.process = (
-					message: ISequencedDocumentMessage,
+				const processMessagesStub = (
```

Comment:
@jatgarg This test is failing with my change. I tried fixing it but it's very hard to understand or change it. It changes the internals of a lot of objects to make the test work. Can you look into fixing this.
I would also suggest to make the test less instrumented because as of now, this is hard to understand / debug and is prone to becoming flaky.